### PR TITLE
[PYG-21] Support generic OIDC

### PR DESCRIPTION
**Pull request by muradsater**
## Description
Please describe the change you have made.

## Checklist:
- [ ] Tests added/updated.
- [ ] Documentation updated.
- [ ] Changelog updated in [CHANGELOG.md](https://github.com/cognitedata/cognite-gql-pygen/blob/main/CHANGELOG.md).
- [x] Version bumped. If triggering a new release is desired, bump the version number in
  [version.py](https://github.com/cognitedata/cognite-gql-pygen/blob/main/cognite/gqlpygen/version.py) and
  [pyproject.toml](https://github.com/cognitedata/cognite-gql-pygen/blob/main/pyproject.toml) per [semantic versioning](https://semver.org/).
- [x] Regenerate example SDK `export PYTHONPATH=. && python dev.py generate`. Need to be run both
  for `pydantic` `v1` and `v2` environments.

### CODE REVIEW

File: cognite/pygen/cli.py [hunk 1]
```diff
@@ -45,7 +46,16 @@ def generate(
...
             tenant_id: str = typer.Option(
-                default=loaded_settings.tenant_id.default, help=loaded_settings.tenant_id.help
+                default=loaded_settings.tenant_id.default, help=loaded_settings.tenant_id.help,
+            ),
+            token_url: str = typer.Option(
+                default=loaded_settings.token_url.default, help=loaded_settings.token_url.help
+            ),
+            scopes: str = typer.Option(
+                default=loaded_settings.scopes.default, help=loaded_settings.scopes.help
+            ),
```
Comment (doctrino) [inline cognite/pygen/cli.py:56]: Do we need this one? 
Can't it just be set as `[f"https://{cdf_cluster}.cognitedata.com/.default"],`?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [tool.poetry]
 name = "cognite-pygen"
-version = "0.27.1"
+version = "0.27.2"
```
Comment (doctrino) [inline pyproject.toml:3]: Worth a major bump, new functionality.

**Review by doctrino — COMMENTED** [review]
The CLI becomes somewhat confessing when there are a lot of arguments and you only need some combination of them. 

What about just having them read from a `toml` and use a `login_flow` variable to make it explicit which one you are using?

Alternatively also support loading from environment variables? (I plan to add that to the SDK)

File: cognite/pygen/cli.py (hunk 1 above)
Comment (muradsater) [reply cognite/pygen/cli.py:56]: Yes, it can be anything when not using Azure AD

**Comment by muradsater** [comment]
> The CLI becomes somewhat confessing when there are a lot of arguments and you only need some combination of them.
> 
> What about just having them read from a `toml` and use a `login_flow` variable to make it explicit which one you are using?

Seems a bit cumbersome to require a toml, but as long as I can use AUTH0 it's fine by me. 

> Alternatively also support loading from environment variables? (I plan to add that to the SDK)

The SDK used to have env variable support, but it was removed a few versions back. You can talk to Erlend on the why I guess.

**Review by doctrino — APPROVED** [review]

**Pull request merged**